components/datetime/src/skeleton/helpers.rs

```diff
@@ -317,6 +319,34 @@ fn adjust_pattern_field_lengths(fields: &[Field], pattern: &mut Pattern) {
     })
 }
 
+/// Alters given Pattern so that it will have a fractional second field if it was requested.
+///
+/// If the requested skeleton included both seconds and fractional seconds and the dateFormatItem
+/// skeleton included seconds but not fractional seconds, then the seconds field of the corresponding
+/// pattern should be adjusted by appending the locale’s decimal separator, followed by the sequence
+/// of ‘S’ characters from the requested skeleton.
+/// (see https://unicode.org/reports/tr35/tr35-dates.html#Matching_Skeletons)
+fn append_fractional_seconds(pattern: &mut Pattern, fields: &[Field]) {
+    if let Some(requested_field) = fields
```

Comment:
thought (follow-up?): For what it's worth, this is iterating through the requested fields another time, and incurs the cost of another loop through the field. This occurs for every pattern lookup. I believe an index to the seconds field could be returned from the skeleton matching algorithm to skip this loop.

The common case here is for there not to be a fractional second, so this would remove a loop from the common case, and potentially save time. I'll leave it up to you if you think this is worth filing as a follow-up.

Reply:
I filed https://github.com/unicode-org/icu4x/issues/1888 for the performance optimizations you suggested. Thanks for the ideas :)